codex/uberdev-codex/lib/run_manifest.py

```
@@ -1361,6 +1361,265 @@ def _secure_open_regular(path: str, flags: int, mode: int = 0o600) -> int:
+            descriptor = _secure_open_regular(
+                candidate, os.O_RDWR | os.O_CREAT | os.O_EXCL, 0o600
+            )
+            return candidate, descriptor
+        except ManifestRuntimeError as exc:
+            if isinstance(exc.__cause__, FileExistsError):
+                continue
+            raise
+    raise ManifestRuntimeError("artifact_attempt_collision")
+
+
+def secure_publish_captured(
```

Ruff (0.15.21)

[warning] Too many branches (18 > 12)

(PLR0912)